### PR TITLE
Add support for cluster role and cluster role binding

### PR DESCRIPTION
**Pull request by Rohit-PX**
Support for ClusterRole and ClusterRoleBinding

Signed-off-by: Rohit-PX <rohit@portworx.com>

### CODE REVIEW

**Comment by Rohit-PX** [comment]
Testing here:  https://jenkins.portworx.co/job/Torpedo/job/tp-dev-test/447/

File: drivers/scheduler/k8s/k8s.go [hunk 1]
```diff
@@ -864,6 +864,42 @@ func (k *K8s) createStorageObject(spec interface{}, ns *v1.Namespace, app *spec.
 
 		logrus.Infof("[%v] Created Role Binding: %v", app.Key, snap.Name)
 		return snap, nil
+	} else if obj, ok := spec.(*rbacv1.ClusterRole); ok {
+		obj.Namespace = ns.Name
+		snap, err := k8sRbac.CreateClusterRole(obj)
```
Comment (mborodin-px) [inline drivers/scheduler/k8s/k8s.go:869]: rename snap variable name as it doesn't related to cluster role object

File: drivers/scheduler/k8s/k8s.go [hunk 2]
```diff
@@ -864,6 +864,42 @@ func (k *K8s) createStorageObject(spec interface{}, ns *v1.Namespace, app *spec.
...
+				App:   app,
+				Cause: fmt.Sprintf("Failed to create Cluster Role: %v. Err: %v", obj.Name, err),
+			}
+		}
+
+		logrus.Infof("[%v] Created Cluster Role: %v", app.Key, snap.Name)
+		return snap, nil
+	} else if obj, ok := spec.(*rbacv1.ClusterRoleBinding); ok {
+		obj.Namespace = ns.Name
+		snap, err := k8sRbac.CreateClusterRoleBinding(obj)
```
Comment (mborodin-px) [inline drivers/scheduler/k8s/k8s.go:887]: the same as above

File: drivers/scheduler/k8s/k8s.go (hunk 1 above)
Comment (Rohit-PX) [reply drivers/scheduler/k8s/k8s.go:869]: sounds good, I am also going to change other objects using the same variable name `snap`

File: drivers/scheduler/k8s/k8s.go [hunk 3]
```diff
@@ -864,6 +864,42 @@ func (k *K8s) createStorageObject(spec interface{}, ns *v1.Namespace, app *spec.
 
 		logrus.Infof("[%v] Created Role Binding: %v", app.Key, snap.Name)
 		return snap, nil
+	} else if obj, ok := spec.(*rbacv1.ClusterRole); ok {
```
Comment (disrani-px) [inline drivers/scheduler/k8s/k8s.go:867]: These all are under `createStorageObject` which doesn't sound right.

**Comment by Rohit-PX** [comment]
https://jenkins.portworx.co/job/Torpedo/job/tp-dev-test/462/

File: drivers/scheduler/k8s/k8s.go [hunk 4]
```diff
@@ -810,13 +837,14 @@ func (k *K8s) createStorageObject(spec interface{}, ns *v1.Namespace, app *spec.
 
 		logrus.Infof("[%v] Created Group snapshot: %v", app.Key, snap.Name)
 		return snap, nil
+
 	} else if obj, ok := spec.(*v1.ServiceAccount); ok {
 		obj.Namespace = ns.Name
-		snap, err := k8sCore.CreateServiceAccount(obj)
+		serviceaccount, err := k8sCore.CreateServiceAccount(obj)
```
Comment (disrani-px) [inline drivers/scheduler/k8s/k8s.go:843]: SA and cluster role need to be removed from this function, right?

File: drivers/scheduler/k8s/k8s.go [hunk 5]
```diff
@@ -2931,6 +2958,161 @@ func (k *K8s) destroyBackupObjects(
...
+			}
+		}
+
+		logrus.Infof("[%v] Created Cluster Role: %v", app.Key, clusterrolebinding.Name)
+		return clusterrolebinding, nil
+	}
+	return nil, nil
+}
+
+func (k *K8s) createNetworkingObjects(
```
Comment (disrani-px) [inline drivers/scheduler/k8s/k8s.go:3062]: I don't see where this is being called from

File: drivers/scheduler/k8s/k8s.go [hunk 6]
```diff
@@ -2931,6 +2958,161 @@ func (k *K8s) destroyBackupObjects(
...
 
+func (k *K8s) createRbacObjects(
+	spec interface{},
+	ns *v1.Namespace,
+	app *spec.AppSpec,
+) (interface{}, error) {
+	if obj, ok := spec.(*appsapi.Deployment); ok {
+		obj.Namespace = ns.Name
+		obj.Spec.Template.Spec.Volumes = k.substituteNamespaceInVolumes(obj.Spec.Template.Spec.Volumes, ns.Name)
+		dep, err := k8sApps.CreateDeployment(obj)
```
Comment (disrani-px) [inline drivers/scheduler/k8s/k8s.go:2969]: Deployments shouldn't be created here

Comment (Rohit-PX) [reply drivers/scheduler/k8s/k8s.go:2969]: Deleted. Was added by mistake.

**Comment by Rohit-PX** [comment]
Tested here:  https://jenkins.portworx.co/job/Torpedo/job/tp-dev-test/469/

**Pull request merged**